examples/pic.cpp

```diff
@@ -0,0 +1,203 @@
+// Copyright 2016 Husky Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <random>
+#include <string>
+#include <vector>
+
+#include "boost/tokenizer.hpp"
+
+#include "core/engine.hpp"
+#include "io/hdfs_manager.hpp"
+#include "io/input/line_inputformat.hpp"
+#include "lib/aggregator_factory.hpp"
+
+typedef std::vector<float> edgeWvec;
+
+class Node {
+    // A node storing node value and connected edges (node value forms the classification vector)
+   public:
+    using KeyT = int;
+
+    Node() {}
+    explicit Node(const KeyT& p) : pos(p), edgeW(), val(0.), valp(0.), valpp(0.) {}
+    Node(const KeyT& p, edgeWvec edgeW, float edgeWsum) {
+        this->pos = p;
+        this->edgeWsum = edgeWsum;
+        this->edgeW = std::move(edgeW);
+        this->val = 0.;
+        this->valp = 0.;
+        this->valpp = 0.;
+    }
+    virtual const KeyT& id() const { return pos; }
+
+    // Serialization and deserialization
+    friend husky::BinStream& operator<<(husky::BinStream& stream, Node u) {
+        stream << u.pos << u.edgeW << u.val << u.valp << u.valpp << u.valtmp;
+        return stream;
+    }
+    friend husky::BinStream& operator>>(husky::BinStream& stream, Node& u) {
+        stream >> u.pos >> u.edgeW >> u.val >> u.valp >> u.valpp >> u.valtmp;
+        return stream;
+    }
+
+    KeyT pos;
+
+    edgeWvec edgeW;
+    float edgeWsum;
+
+    float val;
+    float valp;    // val at previous iteration
+    float valpp;   // val at previousX2 iteration
+    float valtmp;  // val before normalize
```

Comment:
I think the variable valp, valpp, valtmp are not necessary defined in this class, which could take a lot of memory in the runtime. You may refer to the implementation in the bitbucket for more details.